packages/api/models/rules.js

```diff
@@ -159,16 +159,25 @@ class Rule extends Manager {
           return (mapping.EventSourceArn === item.rule.value);
         });
       if (mappingExists) {
-        item.rule.arn = mappingExists.UUID;
-        return item;
+        const mappingEnabled = listData.EventSourceMappings
+          .find((mapping) => { // eslint-disable-line arrow-body-style
+            return (mapping.EventSourceArn === item.rule.value &&
+                    mapping.State === 'Enabled');
+          });
+
+        if (mappingEnabled) {
+          item.rule.arn = mappingEnabled.UUID;
+          return item;
+        }
+        await aws.lambda().deleteEventSourceMapping({ UUID: mappingExists.UUID }).promise();
       }
     }
 
     // create event source mapping
     const params = {
       EventSourceArn: item.rule.value,
       FunctionName: process.env.kinesisConsumer,
-      StartingPosition: 'LATEST',
+      StartingPosition: 'TRIM_HORIZON',
```

Comment:
What's the reason for this change?

Reply:
This was the hardest thing to figure out. When I added a stream setup and teardown. Even after ensuring the stream was 'ACTIVE' and the eventSourceMapping was in place and 'ENABLED', the first message on the stream never triggered the kinesisConsumer.   

I struggled trying to figure out what needed to be in place, and originally set this to be `StartingPosition: 'AT_TIMESTAMP'` with a timestamp of now()  which kind of made sense to me. That would mean that any message placed on the stream after the eventmapping was created would be processed, but alireza said it should be TRIM_HORIZON to process everything on the stream.

Reply:
Hmm I wonder if this some sort of bug in the way AWS configures the lambda trigger when starting position is `LATEST` - like it doesn't know what "LATEST" is until there is at least one record having been on the stream. But `TRIM_HORIZON` makes sense.

Reply:
" like it doesn't know what "LATEST" is until there is at least one record having been on the stream" . I wondered this exactly but never got around to testing it.